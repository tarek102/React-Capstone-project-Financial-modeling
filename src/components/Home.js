import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { BsArrowRightCircle } from 'react-icons/bs';
import { NavLink } from 'react-router-dom';
import { cryptoThunk, searchCrypto } from '../redux/crypto/cryptoReducer';
import SearchBar from './Search';

const Home = () => {
  const crypto = useSelector((state) => state.crypto);
  const dispatch = useDispatch();

  const crypt = crypto.crypto;

  useEffect(() => {
    dispatch(cryptoThunk());
  }, []);

  const [value, setValue] = useState({ search: '' });

  const handleSearch = (e) => {
    e.preventDefault();
    if (value.search === '') {
      return;
    }

    dispatch(searchCrypto(value.search));
    // console.log(value.search);
    setValue({ search: '' });
  };

  const handleChange = (e) => {
    setValue({ search: e.target.value });
    // console.log(value);
  };

  return (
    <section className="wrapper">
      <SearchBar
        handleSearch={handleSearch}
        handleChange={handleChange}
        searchValue={value.search}
      />
      <div className="row cryptos d-flex justify-content-center">
        {crypt.map((coin) => (
          <div className="currencies col-lg-3 col-md-4 mb-5 p-4" key={coin.symbol} id={coin.id}>
            <h2 className="mb-5">{coin.symbol}</h2>
            <div className="d-flex">
              <div>
                <span>
                  Count:
                  {coin.count}
                </span>
                {' '}
                <br />
                <span>
                  Price change:
                  {coin.priceChange}
                </span>
                <br />
                <span>
                  Last price:
                  {coin.lastPrice}
                </span>
              </div>
              <NavLink
                className="align-self-center arrow"
                to={`/details/${coin.symbol}`}
                state={coin}
              >
                <BsArrowRightCircle className="details" />
              </NavLink>
            </div>
          </div>
        ))}
      </div>
    </section>
  );
};

export default Home;
