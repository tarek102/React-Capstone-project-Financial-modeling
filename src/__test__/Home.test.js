import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Home from '../components/Home';
import store from '../redux/ConfigureStore';

describe('Test if Home matches snapshot', () => {
  it('matches', () => {
    const tree = render(
      <Router>
        <Provider store={store}>
          <Home />
        </Provider>
      </Router>,
    );
    expect(tree).toMatchSnapshot();
  });
});
