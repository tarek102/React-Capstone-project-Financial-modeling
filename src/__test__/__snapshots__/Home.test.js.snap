// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if Home matches snapshot matches 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="wrapper d-flex flex-column align-items-center"
      >
        <div
          class="hero"
        >
          <h2>
            Welcome to Crypto traders
          </h2>
          <div
            class="search-div"
          >
            <input
              placeholder="Search coins"
              type="text"
              value=""
            />
            <button
              type="button"
            >
              <a
                class=""
                href="/search"
              >
                Search
              </a>
            </button>
          </div>
        </div>
        <div
          class="row cryptos d-flex flex-row justify-content-center"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="wrapper d-flex flex-column align-items-center"
    >
      <div
        class="hero"
      >
        <h2>
          Welcome to Crypto traders
        </h2>
        <div
          class="search-div"
        >
          <input
            placeholder="Search coins"
            type="text"
            value=""
          />
          <button
            type="button"
          >
            <a
              class=""
              href="/search"
            >
              Search
            </a>
          </button>
        </div>
      </div>
      <div
        class="row cryptos d-flex flex-row justify-content-center"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
